runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/TemplatesSection/TemplatesSection.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FC, useState, useMemo, useEffect, useRef } from 'react';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import Typography from '#src-landing/components/Typography';
+import { TEMPLATE_SECTION_ID } from '#src-landing/utils/utils';
+
+import Categories from './Categories';
+import SearchBar from './SearchBar';
+import styles from './TemplatesSection.module.scss';
+import { categoriesNames, templates } from './TemplatesSection.utils';
+import TemplateTile from './TemplateTile';
+
+const TemplatesSection: FC = () => {
+    const { translate } = useTranslations();
+    const [selectedCategory, setSelectedCategory] = useState<string>(categoriesNames.CATEGORY_ALL);
+    const [searchValue, setSearchValue] = useState<string>('');
+    const gridRef = useRef<HTMLDivElement>(null);
+    const [maxDisplayedTiles, setMaxDisplayedTiles] = useState<number>(6);
+    
+    const filteredTemplates = useMemo(() => templates
+        .filter(template => selectedCategory === 'All' ? true : template.categories.includes(selectedCategory))
+        .filter(template => template.title.toLowerCase().includes(searchValue.toLowerCase()) 
+        || template.description.toLowerCase().includes(searchValue.toLowerCase())), [selectedCategory, searchValue]);
```

Comment:
- move dependencies array to new line
- move template.$VAR.toLowerCase().includes(searchValue.toLowerCase())) to a function

Reply:
done